src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -3181,96 +3171,59 @@ ss::future<checked<tm_transaction, tx_errc>> tx_gateway_frontend::get_tx(
 
 ss::future<checked<tm_transaction, tx_errc>>
 tx_gateway_frontend::get_ongoing_tx(
-  model::term_id expected_term,
+  model::term_id term,
   ss::shared_ptr<tm_stm> stm,
   model::producer_identity pid,
   kafka::transactional_id tx_id,
   model::timeout_clock::duration timeout) {
-    auto tx_opt = co_await stm->get_tx(tx_id);
-    if (!tx_opt.has_value()) {
-        auto status = tx_opt.error();
-        tx_errc err = tx_errc::invalid_producer_id_mapping;
-        if (status == tm_stm::op_status::not_leader) {
-            err = tx_errc::leader_not_found;
-        } else if (status == tm_stm::op_status::unknown) {
-            err = tx_errc::unknown_server_error;
-        } else if (status == tm_stm::op_status::timeout) {
-            err = tx_errc::timeout;
+    auto r0 = co_await get_tx(term, stm, tx_id, timeout);
+    if (!r0.has_value()) {
+        co_return r0.error();
+    }
+
+    auto tx = r0.value();
+
+    if (term != tx.etag) {
+        // very unlikely situation happens only when !is_fetch_tx_supported()
```

Comment:
is_fetch_tx_supported() should always be true going forward, right? (Given our upgrade requirements) Shouldn't a pre-req for this patch be to get rid of all `!is_fetch_tx_supported()`  code paths in a separate commit?  The code is too complex IMO.

Reply:
this a consistency fix which will be backported to 23.1 and there may be update from pre is_fetch_tx_supported; will be able to get rid of this in 23.2